addons/mail/models/mail_notification.py

```diff
@@ -51,6 +54,28 @@ class MailNotification(models.Model):
          'Customer is required for inbox / email notification'),
     ]
 
+    # ------------------------------------------------------------
+    # COMPUTE
+    # ------------------------------------------------------------
+
+    @api.depends('mail_message_id.model', 'mail_message_id.res_id', 'res_partner_id')
+    def _compute_follower_id(self):
+        followed_refs = {(res['partner_id'][0], res['res_model'], res['res_id']): res['id']
```

Comment:
I find it's a bit hard to read will all those indentation (pyramid code)

maybe if we create some variables it takes a bit more lines of code, but it's easier to understand what happen
```py
values = (
    (record.res_partner_id.id, msg.model, msg.res_id)
    for record in self
    for msg in record.mail_message_id
    if msg.res_id and msg.model
)
domain = expression.OR(
    [('partner_id', '=', partner_id), ('res_model', '=', model),
     ('res_id', 'in', [res_id for *__, res_id in values])]
    for (partner_id, model), values in groupby(values, key=lambda r: r[:2])
)
followed_refs = {
    (res['partner_id'][0], res['res_model'], res['res_id']): res['id']
    for res in self.env['mail.followers'].sudo().search_read(
        domain, ['partner_id', 'res_model', 'res_id'])
}
for record in self:
    msg = record.mail_message_id
    record.follower_id = followed_refs.get((record.res_partner_id.id, msg.model, msg.res_id), False)
```

what do you think ?

Reply:
Yes thanks for the suggestion; it is better. I have updated the code with it with 2 very small changes:
- for (partner_id, model), values →  for (partner_id, model), grouped_values: the goal is to not have twice values
- ('res_id', 'in', [res_id for *__, res_id in values]) → list({res_id for *__, res_id in grouped_values})): to remove duplicate because we can have multiple message about the same document.